src/Proxy/External/DocumentProxy.cs

```diff
@@ -0,0 +1,58 @@
+﻿namespace Linn.Stores2.Proxy.External
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Net;
+    using System.Threading;
+    using Linn.Common.Configuration;
+    using Linn.Common.Proxy;
+    using Linn.Common.Serialization.Json;
+    using Linn.Stores2.Domain.LinnApps.External;
+    using Linn.Stores2.Domain.LinnApps.Requisitions;
+    using Linn.Stores2.Resources.External;
+
+    public class DocumentProxy : IDocumentProxy
+    {
+        private readonly IRestClient restClient;
+
+        public DocumentProxy(IRestClient restClient)
+        {
+            this.restClient = restClient;
+        }
+
+        public DocumentResult GetCreditNote(int documentNumber, int? documentLine)
+        {
+            DocumentResult result = null;
+
+            var uri = $"{ConfigurationManager.Configuration["PROXY_ROOT"]}/sales/credit-notes/{documentNumber}";
+
+            var response = this.restClient.Get(CancellationToken.None, new Uri(uri, UriKind.RelativeOrAbsolute), new Dictionary<string, string>(), this.JsonHeaders()).Result;
```

Comment:
This can be awaited (as opposed to .Result which forces synchronous run)

Reply:
my original version of DocumentProxy did make it all async but I had trouble making it work is there another example of an async RestClient type proxy?